File: test/java/test/PureClientInfo.java
package test;

import java.io.Serializable;
import java.util.List;

public class PureClientInfo implements Serializable {

    private static final long serialVersionUID = -4839365452784671213L;
    private String serverIP; // configServer IP
    private String dataId;
    private String clientId;
    private String hostId;
    private List<String> groups;
    private boolean isValid;

    /** Checking whether group is exists in groups. */
    public boolean isInGroup(String groupId) {
        return groups.contains(groupId);
    }

    public String getDataId() {
        return dataId;
    }

    public void setDataId(String dataId) {
        this.dataId = dataId;
    }

    public String getClientId() {
        return clientId;
    }

    public void setClientId(String clientId) {
        this.clientId = clientId;
    }

    public String getHostId() {
        return hostId;
    }

    public void setHostId(String hostId) {
        this.hostId = hostId;
    }

    public List<String> getGroups() {
        return groups;
    }

    public void setGroups(List<String> groups) {
        this.groups = groups;
    }

    public void setServerIP(String ip) {
        serverIP = ip;
    }
    public String getServerIP() {
        return serverIP;
    }

	public void setValid(boolean isValid) {
		this.isValid = isValid;
	}

	public boolean isValid() {
		return isValid;
	}

    @Override
    public String toString() {
        return new StringBuilder("dataId=").append(dataId).append(" clientId=").append(clientId).append(" hostId=").append(hostId).append(
                " groups=").append(groups).append(" isValid=").append(isValid).toString();
    }



}

File: test/fixtures/in/object/PureClientInfo.js
module.exports = { '$':
   { isValid: true,
     clientId: 'some-clientId',
     dataId: 'some-dataId',
     groups:
      { '$': { size: 2, capacity: 10 },
        '$class':
         { name: 'java.util.ArrayList',
           serialVersionUID: '8683452581122892189',
           flags: 3,
           fields: [ { type: 'I', name: 'size' } ],
           superClass: null },
        '_$': [ 'SOFA-GROUP', 'HSF' ] },
     hostId: '127.0.0.1',
     serverIP: '127.0.0.2' },
  '$class':
   { name: 'javaio.test.PureClientInfo',
     serialVersionUID: '-4839365452784671213',
     flags: 2,
     fields:
      [ { type: 'Z', name: 'isValid' },
        { type: 'L', name: 'clientId', classname: 'Ljava/lang/String;' },
        { type: 'L', name: 'dataId', classname: 'Ljava/lang/String;' },
        { type: 'L', name: 'groups', classname: 'Ljava/util/List;' },
        { type: 'L', name: 'hostId', classname: 'Ljava/lang/String;' },
        { type: 'L', name: 'serverIP', classname: 'Ljava/lang/String;' } ],
     superClass: null } }